cpp/CMakePresets.json

```diff
@@ -1,165 +1,231 @@
+        "ARROW_BUILD_INTEGRATION": "OFF",
+        "ARROW_BUILD_TESTS": "OFF",
+        "ARROW_EXTRA_ERROR_CONTEXT": "OFF",
+        "CMAKE_BUILD_TYPE": "Release"
+      }
+    },
+    {
+      "name": "ninja-release",
+      "inherits": "ninja-debug",
+      "displayName": "Release build with more optional components",
+      "cacheVariables": {
+        "ARROW_BUILD_INTEGRATION": "OFF",
+        "ARROW_BUILD_TESTS": "OFF",
+        "ARROW_EXTRA_ERROR_CONTEXT": "OFF",
+        "CMAKE_BUILD_TYPE": "Release"
+      }
+    },
+    {
+      "name": "ninja-release-cuda",
+      "inherits": "ninja-debug-cuda",
+      "displayName": "Release build with CUDA integration",
+      "cacheVariables": {
+        "ARROW_BUILD_INTEGRATION": "OFF",
+        "ARROW_BUILD_TESTS": "OFF",
+        "ARROW_EXTRA_ERROR_CONTEXT": "OFF",
+        "CMAKE_BUILD_TYPE": "Release"
+      }
+    },
+    {
+      "name": "ninja-release-flight",
+      "inherits": "ninja-debug-flight",
+      "displayName": "Release build with Flight",
+      "cacheVariables": {
+        "ARROW_BUILD_INTEGRATION": "OFF",
+        "ARROW_BUILD_TESTS": "OFF",
+        "ARROW_EXTRA_ERROR_CONTEXT": "OFF",
+        "CMAKE_BUILD_TYPE": "Release"
+      }
+    },
+    {
+      "name": "ninja-release-gandiva",
+      "inherits": "ninja-debug-gandiva",
+      "displayName": "Release build with Gandiva",
+      "cacheVariables": {
+        "ARROW_BUILD_INTEGRATION": "OFF",
+        "ARROW_BUILD_TESTS": "OFF",
+        "ARROW_EXTRA_ERROR_CONTEXT": "OFF",
+        "CMAKE_BUILD_TYPE": "Release"
+      }
+    },
+    {
+      "name": "ninja-release-python",
+      "inherits": "ninja-debug-python",
+      "displayName": "Release build with Python support",
+      "cacheVariables": {
+        "ARROW_BUILD_INTEGRATION": "OFF",
+        "ARROW_BUILD_TESTS": "OFF",
+        "ARROW_EXTRA_ERROR_CONTEXT": "OFF",
+        "CMAKE_BUILD_TYPE": "Release"
+      }
+    },
+    {
+      "name": "ninja-release-maximal",
+      "inherits": "ninja-debug-maximal",
+      "displayName": "Release build with everything enabled (except benchmarks and CUDA)",
+      "cacheVariables": {
+        "ARROW_BUILD_INTEGRATION": "OFF",
+        "ARROW_BUILD_TESTS": "OFF",
+        "ARROW_EXTRA_ERROR_CONTEXT": "OFF",
+        "CMAKE_BUILD_TYPE": "Release"
+      }
+    },
+
+    {
+      "name": "ninja-benchmarks-basic",
+      "inherits": "ninja-release-basic",
+      "displayName": "Benchmarking build with reduced dependencies",
+      "cacheVariables": {
+        "ARROW_BUILD_BENCHMARKS": "ON",
+        "ARROW_BUILD_BENCHMARKS_REFERENCE": "ON",
+        "CMAKE_BUILD_TYPE": "RelWithDebInfo"
+      }
+    },
+    {
+      "name": "ninja-benchmarks",
+      "inherits": "ninja-release",
+      "displayName": "Benchmarking build with more optional components",
+      "cacheVariables": {
+        "ARROW_BUILD_BENCHMARKS": "ON",
+        "ARROW_BUILD_BENCHMARKS_REFERENCE": "ON",
+        "CMAKE_BUILD_TYPE": "RelWithDebInfo"
```

Comment:
inherit from `ninja-benchmarks-basic` and `ninja-release` and we avoid the variables.